query/src/interpreters/interpreter_factory.rs

```diff
@@ -52,7 +52,25 @@ impl InterpreterFactory {
             PlanNode::TruncateTable(v) => TruncateTableInterpreter::try_create(ctx, v),
             PlanNode::UseDatabase(v) => UseDatabaseInterpreter::try_create(ctx, v),
             PlanNode::SetVariable(v) => SettingInterpreter::try_create(ctx, v),
-            PlanNode::InsertInto(v) => InsertIntoInterpreter::try_create(ctx, v),
+            PlanNode::InsertInto(v) => {
+                let select = match v.select_plan.clone().take() {
```

Comment:
Great job 👍 
It would be nice to check and build select plan in the InsertIntoInterpreter, it looks more cleaner